sklearn/decomposition/_online_lda_fast.pyx

```diff
@@ -8,9 +8,11 @@ cnp.import_array()
 from libc.math cimport exp, fabs, log
 from numpy.math cimport EULER
 
+# TODO: use const memory views for fused type "floating" with Cython 3.0,
```

Comment:
Hi @OmarManzoor thank you for this PR!

Unfortunately, setting memory views to `const` is mandatory since read-only arrays will raise an error. See the following example on your branch:

```python
import numpy as np
from sklearn.decomposition._online_lda_fast import mean_change

a = np.random.randn(10)
b = np.random.randn(10)

a.setflags(write=False)
mean_change(a, b)
```

Will result in:
```Traceback
ValueError: buffer source array is read-only
```

So the -Wcpp warning for `_online_lda_fast.pyx` can't be removed just yet.

Reply:
My intuition is that we should wait for Cython >3 to be released but I'll leave that suggestion to @jjerphan

Reply:
That's right: we won't be able to solve before Cython 3.0 is released.

Reply:
I think then we can close this PR for now and update this on the relevant issue?

Reply:
Yes.